images/linux/scripts/installers/docker-moby.sh

```diff
@@ -28,10 +28,18 @@ systemctl is-enabled --quiet docker.service || systemctl enable docker.service
 sleep 10
 docker info
 
+# If credentials are provided, attempt to log into Docker Hub with a paid account to avoid Docker Hub's rate limit.
+if [ "${DOCKERHUB_LOGIN}" ] && [ "${DOCKERHUB_PASSWORD}" ]; then
+  docker login --username "${DOCKERHUB_LOGIN}" --password "${DOCKERHUB_PASSWORD}"
+fi
+
 # Pull images
 images=$(get_toolset_value '.docker.images[]')
 for image in $images; do
     docker pull "$image"
 done
 
-invoke_tests "Tools" "Docker"
\ No newline at end of file
+# Always attempt to logout so we do not leave our credentials on the built image.
+docker logout
```

Comment:
In my testing `docker logout` will not return a non-zero exit code even if you never logged into DockerHub. Therefore I believe it's safe/safer to always call it in this script.